### PR TITLE
[Instrumentation.AspNet] Fix multiple routes of same template in attribute-based routing

### PR DESCRIPTION
**Pull request by tchowice**
## Changes

ASP.NET instrumentation fails to extract route for attribute-based routing, if multiple HTTP methods use the same route template--a common scenario for RESTful APIs. For example:

```csharp
public class WidgetsController : ApiController
{
    [Route("api/widgets/{id}")]
    public string Get(int id) {...}

    [Route("api/widgets/{id}")]
    public void Put(int id, [FromBody]string value) {...}

    [Route("api/widgets/{id}")]
    public void Delete(int id) {...}
}
```

In this case, the `MS_Subroutes` array will have multiple elements, each corresponding to a different HTTP method.

The original code only extracts the route template if there's only one element in the array. The correct thing to do is to just extract the template from the first subroute, since all subroutes have the same template.

It's not possible for MS_Subroutes array to have different templates in its elements. WebAPI would fail to route and throw an exception, if you declare multiple actions for the same method/template

## Merge requirement checklist

* [x] [CONTRIBUTING](https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/main/CONTRIBUTING.md) guidelines followed (license requirements, nullable enabled, static analysis, etc.)
* [x] Unit tests added/updated
* [ ] Appropriate `CHANGELOG.md` files updated for non-trivial changes
* [ ] Changes in public API reviewed (if applicable)

### CODE REVIEW

**Comment by Kielek** [comment]
Please sign EasyCLA. It is hard requirement to accept any contribution to this repository.

**Comment by tchowice** [comment]
> Please sign EasyCLA. It is hard requirement to accept any contribution to this repository.

Thanks. Working with my company to figure this out.

**Comment by tchowice** [comment]
> > Please sign EasyCLA. It is hard requirement to accept any contribution to this repository.
> 
> Thanks. Working with my company to figure this out.

**Comment by tchowice** [comment]
> Please sign EasyCLA. It is hard requirement to accept any contribution to this repository.

Okay, I finally got my company to sign the CLA. Please proceed with reviewing!

**Review by ysolomchenko — COMMENTED** [review]
Add an entry to the Changelog

File: test/OpenTelemetry.Instrumentation.AspNet.Tests/RouteTestHelper.cs [hunk 1]
```diff
@@ -40,6 +41,75 @@ public static HttpContext BuildHttpContext(string url, int routeType, string? ro
...
+                                        },
+                                    },
+                                },
+                            },
+                        },
+                    },
+                };
+                routeData.Values.Add(
+                    "MS_SubRoutes",
+                    multiMethodSubroutes);
```
Comment (ysolomchenko) [inline test/OpenTelemetry.Instrumentation.AspNet.Tests/RouteTestHelper.cs:111]: I think we can simplify to:

```
    routeData = new RouteData();
    var httpMethods = new[] { HttpMethod.Get, HttpMethod.Put, HttpMethod.Delete };

    var multiMethodSubroutes = httpMethods.Select(method => new
    {
        Route = new
        {
            RouteTemplate = routeTemplate,
            DataTokens = new Dictionary<string, object>
            {
                ["actions"] = new[]
                {
                    new
                    {
                        SupportedHttpMethods = new[] { method }
                    }
                }
            }
        }
    }).ToArray();

    routeData.Values.Add("MS_SubRoutes", multiMethodSubroutes);
```

**Review by RassK — APPROVED** [review]

**Review by ysolomchenko — APPROVED** [review]

**Pull request merged**